SalesforceCommerceCloud/pwa-kit
# Add support for inventory availability in product bundles @W-16322576@

# Description

This PR adds support for inventory availability in product bundles



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

1. `git checkout ju/fix-bundles-inventory`
2. replace your `default.js` with the one provided in the slack thread
3. search for 'bundle' in PLP and then select 'Women's Business Casual Bundle'
4. Select variants all products except 'Swing Tank'
5. Select `Color: Black` and `Size: M` for 'Swing Tank
6. Observe `Add Bundle to Cart` button is disabled and `Out of stock for Swing tank` message appears
7. Select `Color: Black` and `Size: L` for 'Swing Tank'
8. Increase parent bundle quantity to 4
9. Observe `Add Bundle to Cart` button is disabled and `Only 3 left for Swing tank!` message appears
10. decrease parent bundle quantity to 3 and add bundle to cart
11. Go to cart page with `View Cart` button
12. increase quantity to 4 in product line item for product bundle on cart page
13. Observe `Only 3 left for Swing Tank!` message appears
14. decrease quantity to 3
15. Click on `Edit` to bring up product bundle edit modal
16. Select `Color: Black` and `Size: M` for 'Swing Tank'
17. Observe 'Update' button is disabled and `Out of stock for Swing tank` message appears
18. Select `Color: Black` and `Size: L` for 'Swing Tank' and increase quantity to 4
19. Observe 'Update' button is disabled and `Only 3 left for Swing tank!` message appears
20. Select another variant for 'Swing Tank' that is not size M or L and click 'Update'
21. Observe that the product item has updated as expected

### ** NOTE: PLEASE DO NOT CHECKOUT WITH THE BUNDLE AS IT WILL AFFECT INVENTORY**

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
